www/src/components/shared/egghead-embed.js

```diff
@@ -1,25 +1,45 @@
-import React, { Fragment } from "react"
+import React, { useRef, useState, useCallback, useLayoutEffect } from "react"
 import PropTypes from "prop-types"
 
-const EggheadEmbed = ({ lessonLink, lessonTitle }) => (
-  <Fragment>
-    <iframe
-      className="egghead-video"
-      width={600}
-      height={348}
-      src={`${lessonLink}/embed`}
-      title={`Video: ${lessonTitle}`}
-      allowFullScreen
-    />
-
-    <p>
-      <em>
-        Video hosted on <a href={lessonLink}>egghead.io</a>
-      </em>
-      .
-    </p>
-  </Fragment>
-)
+const EggheadEmbed = ({ lessonLink, lessonTitle }) => {
+  const [iframeWidth, setIframeWidth] = useState(0)
+  const iframeRef = useRef()
+
+  const handleResize = useCallback(
+    () => setIframeWidth(iframeRef.current.clientWidth),
+    [iframeRef.current]
+  )
+
+  useLayoutEffect(() => {
+    handleResize()
+    window.addEventListener(`resize`, handleResize)
+
+    return () => {
+      window.removeEventListener(`resize`, handleResize)
+    }
+  }, [iframeRef.current])
+
+  return (
+    <>
+      <iframe
+        ref={iframeRef}
+        className="egghead-video"
+        width={600}
+        height={iframeWidth * 0.57}
```

Comment:
Why `0.57` specifically? 🤔 

I assume that's the aspect ratio of the videos generally? What if that changes?

Reply:
@sidharthachatterjee yes, it is the ratio of the height to the width of the video element inside the iframe, which is fixed. I think that it won't change as it's a native ratio of the video. Maybe it's a good idea to create a `constant` variable on the top of this component, so other users will understand what this number stands for? What do you think about this?

Reply:
Yup, could definitely be a constant as opposed to a magic number

Reply:
@sidharthachatterjee commit with this improvement added :)

Reply:
@qmixi Could also just be `0.5625` or `9/16` :)